src/HotChocolate/ApolloFederation/src/ApolloFederation/Helpers/EntitiesResolver.cs

```diff
@@ -11,39 +11,79 @@ namespace HotChocolate.ApolloFederation.Helpers;
 /// </summary>
 internal static class EntitiesResolver
 {
-    public static async Task<List<object?>> ResolveAsync(
+    public static async Task<IReadOnlyList<object?>> ResolveAsync(
```

Comment:
@michaelstaib  as this a internal resolver i changed it to List. we save a list allocation by doing this.